### ISSUE TITLE
Grad w.r.t. nufft coordinates after `jax.vmap` throws an error

### ISSUE THREAD

**Issue by Matematija** [flatironinstitute/jax-finufft#54]
Hi!

I think I found another potential bug. I get an error whenever I need to take a derivative w.r.t. to input coordinates of `nufft1` and there is a `jax.vmap` in my code. It looks like some internal array reshapes somehow don't interact well with `vmap`.

Minimal reproducer:

```python
from functools import partial
import numpy as np
import jax
from jax import numpy as jnp
from jax_finufft import nufft1

@partial(jax.vmap, in_axes=(0, 0))
def aux(f, x): # Some dummy scalar output function
    f_ = f.astype(jnp.complex64)
    x_ = x.T.astype(jnp.float32)
    f_hat = nufft1((32, 32, 32), f_, *x_, iflag=-1)
    return jnp.real(f_hat).mean()

def test(f, x):
    return aux(f, x).mean()

f = np.random.randn(8, 1000)
x = np.random.randn(8, 1000, 3)

print(test(f, x).shape) # Works OK.
print( jax.grad(test, argnums=1)(f, x).shape ) # Blows up, but only if argnums=1
```

The same error is thrown on both CPU and GPU.

**Comment by lgarrison** [comment]
This is the error I get. Agreed it looks like something with the reshaping, any ideas @dfm?

```
(venv-23) scclin021:~/jax-finufft$ python scripts/gh54.py 
()
Traceback (most recent call last):
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 20, in <module>
    print(test(f, x).shape) # Works OK.
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 15, in test
    return aux(f, x).mean()
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 11, in aux
    f_hat = nufft1((32, 32, 32), f_, *x_, iflag=-1)
  File "/mnt/home/lgarrison/jax-finufft/venv-23/lib/python3.10/site-packages/jax_finufft/ops.py", line 33, in nufft1
    result = nufft1_p.bind(
jax._src.source_info_util.JaxStackTraceBeforeTransformation: TypeError: reshape total size must be unchanged, got new_sizes (8, 1, 1, 32, 32, 32) for shape (8, 8, 32, 32, 32).

The preceding stack trace is the source of the JAX operation that, once transformed by JAX, triggered the following exception.

--------------------

The above exception was the direct cause of the following exception:

jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 21, in <module>
    print( jax.grad(test, argnums=1)(f, x).shape ) # Blows up, but only if argnums=1
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 15, in test
    return aux(f, x).mean()
  File "/mnt/home/lgarrison/jax-finufft/scripts/gh54.py", line 11, in aux
    f_hat = nufft1((32, 32, 32), f_, *x_, iflag=-1)
TypeError: reshape total size must be unchanged, got new_sizes (8, 1, 1, 32, 32, 32) for shape (8, 8, 32, 32, 32).
```

**Comment by dfm** [comment]
I'll take a look in the next day or two!

**Comment by Matematija** [comment]
Thank you!

**Comment by dfm** [comment]
Fixed by #54, I believe! Please feel free to re-open if it's not!
